rskj-core/src/main/java/co/rsk/core/bc/MiningMainchainViewImpl.java

```diff
@@ -19,102 +19,200 @@
 package co.rsk.core.bc;
 
 import co.rsk.crypto.Keccak256;
-import org.ethereum.core.Block;
-import org.ethereum.core.Blockchain;
+import org.ethereum.core.BlockHeader;
+import org.ethereum.db.BlockStore;
 
 import javax.annotation.concurrent.GuardedBy;
 import java.util.*;
-import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class MiningMainchainViewImpl implements MiningMainchainView {
```

Comment:
Can we add @ThreadSafe  annotation? , is just to let the world know that we were worried about concurrency